### PR TITLE
Master

### PR DESCRIPTION
**Pull request by Wasif-Shahzad**
pull request as woo asked

### CODE REVIEW

File: lib/util/add_task_dialogue_box.dart [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import 'package:flutter/material.dart';
+import 'package:todo_app/util/dialogue_box_button.dart';
+
+class DialogueBox extends StatelessWidget {
+  final controller;
```
Comment (brainwo) [inline lib/util/add_task_dialogue_box.dart]: I believe your linter will yell at you for not having explicit type.
This should be
```dart
final TextEditingController controller;
```
instead

File: lib/main.dart [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      ),
+      home: const HomePage(),
+      routes: {
+        '/monday': (context) => const HomePage(),
+        '/tuesday': (context) => const TuesdayTasks(),
+        '/wednesday': (context) => const WednesdayTasks(),
+        '/thursday': (context) => const ThursdayTasks(),
+        '/friday': (context) => const FridayTasks(),
+        '/saturday': (context) => const SaturdayTasks(),
+        '/sunday': (context) => const SundayTasks(),
```
Comment (brainwo) [inline lib/main.dart]: You only change `day` and `dayIndex` so you should be able to just reuse `HomePage`
```dart
'/monday': (context) => const HomePage(day 'monday', dayIndex: 0),
'/tuesday': (context) => const HomePage(day: 'tuesday', dayIndex: 1),
```

File: lib/pages/friday_tasks.dart
Comment (brainwo) [inline lib/pages/friday_tasks.dart:1]: This file is redundant

File: lib/pages/saturday_tasks.dart
Comment (brainwo) [inline lib/pages/saturday_tasks.dart:1]: This file is redundant

File: lib/pages/sunday_tasks.dart
Comment (brainwo) [inline lib/pages/sunday_tasks.dart:1]: This file is redundant

File: lib/pages/thursday_tasks.dart
Comment (brainwo) [inline lib/pages/thursday_tasks.dart:1]: This file is redundant

File: lib/pages/tuesday_tasks.dart
Comment (brainwo) [inline lib/pages/tuesday_tasks.dart:1]: This file is redundant

File: lib/pages/wednesday_tasks.dart
Comment (brainwo) [inline lib/pages/wednesday_tasks.dart:1]: This file is redundant

File: lib/util/todo_tile.dart [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        child: Container(
+          padding: const EdgeInsets.all(14.0),
+          child: Row(
+            children: [
+              Checkbox(value: taskStatus, onChanged: onChanged),
+              const SizedBox(width: 10),
+              Text(
+                taskname,
+                style: TextStyle(
+                  decoration: taskStatus ? TextDecoration.lineThrough : TextDecoration.none,
```
Comment (brainwo) [inline lib/util/todo_tile.dart]: This code is too long (dart maximum is 80 characters wide), your formatter should have already split the lines

File: lib/util/add_task_dialogue_box.dart [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+          children: [
+            TextField(
+              autocorrect: true,
+              controller: controller,
+              decoration: InputDecoration(
+                border: OutlineInputBorder(borderRadius: BorderRadius.circular(15.0),),
+                hintText: "Enter the task...",
+
```
Comment (brainwo) [inline lib/util/add_task_dialogue_box.dart]: Why is this not formatted? You should enable format-on-save in your editor.

File: lib/pages/home_page.dart [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+              child: const Text(
+                'Days',
+                style: TextStyle(fontSize: 24,),
+              )
+            ),
+            ListTile(
+              title: const Text('Monday'),
+              onTap: (){
+                Navigator.pop(context);
+                Navigator.pushNamed(context, '/monday');
```
Comment (brainwo) [inline lib/pages/home_page.dart]: You can use pushReplacement to replace
https://api.flutter.dev/flutter/widgets/Navigator/pushReplacement.html

**Review by brainwo — CHANGES_REQUESTED** [review]
This is not everything, but at least this is the first suggestion I could give to you

File: lib/pages/home_page.dart (hunk 5 above)
Comment (Wasif-Shahzad) [reply lib/pages/home_page.dart]: everything done sir. 👍

**Review by brainwo — APPROVED** [review]
I think it's good enough, though I have some stylistic preferences to use arrow function in a function with only one line:
```dart
onTap: () {
  Navigator.pushReplacementNamed(context, '/monday');
},
```
to:
```dart
onTap: () =>
  Navigator.pushReplacementNamed(context, '/monday'),
```
Keep up the good work 👍

**Comment by brainwo** [comment]
Side note: you can switch your main branch back to `master`

**Comment by Wasif-Shahzad** [comment]
> Side note: you can switch your main branch back to `master`

thanks for your review and help

**Pull request merged**